### DISCUSSION TITLE
Minor issue with virtual switches

### DISCUSSION THREAD

**Discussion by OlwinFroon** [mak-gitdev/HA_enoceanmqtt#108] [Q&A]
I defined a bunch of virtual switches like this:
```
[virtual_remote_1]
address = 0xffffffff
rorg = 0xF6
func = 0x02
type = 0x01
sender = 0x(base address + 1)
virtual = 1
```

There are supposed to be two ways to emulate pressing, say, the AI button:

a) by selecting action "AI"
- wait for a bit, then 
- select "none"

b) by setting the "AI" switch to "on"
- hit "send"
- wait for a bit, then
- set the "AI" switch of "off" again and
- hit "send"

a) works as expected, it'd send 
Select "AI" sends 0xF6 0x10 [address] 0x30
Select "none" sends 0xF6 0x00 [address] 0x20

b) does not.
It always sends sends 0xF6 0x00 [address] 0x20 regardless the switch settings.

This is not an immediate issue for me, because I don't need to emulate multi button pressing.

**Comment by OlwinFroon** [comment]
Hi @mak-gitdev,

I wrote a bunch of scripts that'd 
- change a v-switch's action to AI, AO, BI or BO, depending on what I need them to do
- wait for 300 msec
- change action to none

So far, this works nicely.

The only _issue_ here is just the log file complaining that the v-switches would be unknown sensors:

```
2024-01-12 09:37:31,580 INFO: sending: 11:22:33:82->FF:FF:FF:FF (-255 dBm): 0x01 
['0xf6', '0x30', '0x11', '0x22', '0x33', '0x82', '0x30'] 
['0x3', '0xff', '0xff', '0xff', '0xff', '0xff', '0x0'] 
2024-01-12 09:37:31,643 INFO: Sending packet
2024-01-12 09:37:31,683 INFO: got response packet: OK
2024-01-12 09:37:31,784 INFO: received: 11:22:33:82->FF:FF:FF:FF (-85 dBm): 0x01 
['0xf6', '0x30', '0x11', '0x22', '0x33', '0x82', '0x31'] 
['0x0', '0xff', '0xff', '0xff', '0xff', '0x55', '0x0'] 

2024-01-12 09:37:31,785 INFO: unknown sensor: 11:22:33:82

2024-01-12 09:37:31,884 INFO: sending: 11:22:33:82->FF:FF:FF:FF (-255 dBm): 0x01 
['0xf6', '0x0', '0x11', '0x22', '0x33', '0x82', '0x20'] 
['0x3', '0xff', '0xff', '0xff', '0xff', '0xff', '0x0'] 
2024-01-12 09:37:31,885 INFO: Sending packet
2024-01-12 09:37:31,924 INFO: got response packet: OK
2024-01-12 09:37:32,025 INFO: received: 11:22:33:82->FF:FF:FF:FF (-83 dBm): 0x01 
['0xf6', '0x0', '0x11', '0x22', '0x33', '0x82', '0x21'] 
['0x0', '0xff', '0xff', '0xff', '0xff', '0x53', '0x0'] 

2024-01-12 09:37:32,026 INFO: unknown sensor: 11:22:33:82
```

**Reply by mak-gitdev** [reply]
Hi @OlwinFroon, 

Do you have repeaters in your EnOcean network ?
That could explain why you got such messages, and if so, not an issue at all 🙂

**Reply by OlwinFroon** [reply]
I do. And yes, it's more like a minor annoyance.

Still, in this case, 11:22:33:82 is NOT an unknown sensor and I'm afraid that all repeater messages might be treated in a similar way, clogging up the log and making it harder to find real problems.

**Reply by mak-gitdev** [reply]
Hi @OlwinFroon,

I think you can add in your device file:
```
[unwanted_device]
address = 0x11223382
ignore = 1
```

Please let me know if that fix your issue :+1:

**Reply by OlwinFroon** [reply]
You mean ignore the virtual switch? Man, that feels wrong :)

Whatever, I just did that and, at least, everything still works.

For some reasons, though, no telegrams go through the repeater right now.
Need to take a closer look at that next week.

**Reply by mak-gitdev** [reply]
Hi @OlwinFroon,

> You mean ignore the virtual switch? Man, that feels wrong :)

:rofl: We only ignore the Rx data from the virtual switch (from repeaters). Sending is not impacted :wink:

> For some reasons, though, no telegrams go through the repeater right now.

I didn't understand this one. What do you mean ?

**Reply by OlwinFroon** [reply]
Means, my repeater doesn't work anymore.
I need to check why.

**Reply by mak-gitdev** [reply]
How do you know that your repeater doesn't work ?
If it is because you don't see data from the repeater in your log for this virtual switch, then this is normal as this is the purpose of the `ignore` option.
Otherwise, note that adding `ignore` to the device configuration should not impact the EnOcean network itself in any manner.
It is just a filter for your log file.
So let you see why this happens then. Do not hesitate if can help :slightly_smiling_face:

**Comment by OlwinFroon** [comment]
Right. Facepalm.

The repeater is working fine and so is the workaround.

Thank you!
